tools.py

```diff
@@ -32,41 +37,74 @@ def subset(input_path, output_path, audio_name, align_boundary_words, cer, wer,
+	audio_path_res = []
+	prev_audio_path = ''
+	for t in audio_transcripts:
 		audio_path = t['audio_path']
-		signal = audio.read_audio(audio_path, sample_rate, normalize = False, backend = audio_backend)[0] if audio_path != prev_audio_path else signal
+		signal = audio.read_audio(audio_path, sample_rate, normalize=False, backend=audio_backend)[
+			0] if audio_path != prev_audio_path else signal
+
+		if signal.shape[-1] == 0: # bug with empty audio files witch produce empty cut file
+			print('Empty audio_path ', audio_path)
+			return []
+
 		t['channel'] = 0 if len(signal) == 1 else None if mono else t.get('channel')
-		segment = signal[slice(t['channel'], 1 + t['channel']) if t['channel'] is not None else ..., int(max(t['begin'] - dilate, 0) * sample_rate) : int((t['end'] + dilate) * sample_rate)]
-		segment_path = os.path.join(output_path, os.path.basename(audio_path) + '.{channel}-{begin:.06f}-{end:.06f}.wav'.format(**t))
-		audio.write_audio(segment_path, segment, sample_rate, mono = True)
+		segment = signal[slice(t['channel'], 1 + t['channel']) if t['channel'] is not None else ...,
+		int(max(t['begin'] - dilate, 0) * sample_rate): int((t['end'] + dilate) * sample_rate)]
+
+		segment_file_name = os.path.basename(audio_path) + '.{channel}-{begin:.06f}-{end:.06f}.wav'.format(**t)
+		digest = hashlib.md5(segment_file_name.encode('utf-8')).hexdigest()
+		sub_path = [digest[-1:], digest[:2]] if add_sub_paths else []
+		segment_path_components = [output_path] + sub_path + [segment_file_name]
+
+		segment_path = os.path.join(*segment_path_components)
+		os.makedirs(os.path.join(*segment_path_components[:-1]), exist_ok=True)
+		audio.write_audio(segment_path, segment, sample_rate, mono=True)
+
 		if strip_prefix:
-			segment_path = segment_path[len(strip_prefix):] if segment_path.startswith(strip_prefix) else segment_path
-			t['audio_path'] = t['audio_path'][len(strip_prefix):] if t['audio_path'].startswith(strip_prefix) else t['audio_path']
-
-		t = dict(audio_path = segment_path,
-				channel = 0 if len(signal) == 1 else None,
-				begin = 0.0,
-				end = segment.shape[-1] / sample_rate,
-				speaker = t.pop('speaker', None),
-				ref = t.pop('ref', None),
-				hyp = t.pop('hyp', None),
-				cer = t.pop('cer', None),
-				alignment = t.pop('alignment', []),
-				words = t.pop('words', []),
-				meta = t)
+			segment_path = segment_path[len(strip_prefix):] if segment_path.startswith(
+					strip_prefix) else segment_path
+			t['audio_path'] = t['audio_path'][len(strip_prefix):] if t['audio_path'].startswith(strip_prefix) else \
+			t['audio_path']
+
+		t = dict(audio_path=segment_path,
+				audio_name=os.path.basename(segment_path),
+				channel=0 if len(signal) == 1 else None,
+				begin=0.0,
+				end=segment.shape[-1] / sample_rate,
+				speaker=t.pop('speaker', None),
+				ref=t.pop('ref', None),
+				hyp=t.pop('hyp', None),
+				cer=t.pop('cer', None),
+				wer=t.pop('wer', []),
```

Comment:
Please fix the default value

Reply:
Your previous request was "replace default with []"  https://github.com/vadimkantorov/convasr/pull/6#discussion_r428561423

Reply:
Sorry, my previous comment was a misclick.

What are types of `wer` and `alignment`? `wer` is a float scalar or absent. `alignment` is a (potentially empty) list, hence the default types